lua/fzf.lua

```diff
@@ -32,14 +32,65 @@ local function get_lines_from_file(file)
   return t
 end
 
+-- can be overwritten by the user
+FZF.default_window_options = {}
+FZF.default_options = {
+  fzf_binary      = "fzf",
+  fzf_cli_args    = "",
+}
+
+-- for convenience window functions
+-- currently adds a border by default
+local function process_options(user_opts, window_options)
+
+  -- backward compatibility:
+  -- 'user_opts' used to be 'fzf_cli_args'
+  if type(user_opts) == "string" then
+    local fzf_cli_args = user_opts
+    user_opts = { fzf_cli_args = fzf_cli_args }
+  end
+
+  if not user_opts then user_opts = {} end
+  if not user_opts.window_options then user_opts.window_options = {} end
+
+  -- backward compatibility:
```

Comment:
I think my comment above would tame some of the complexity with this approach. Backward compatibility would be "built-in". Open to suggestions!

Reply:
It’s a good idea, but say someone in the past decided to set `default_window_options` we’d still need to copy it over into the `user_opts` no?

Whatever you make of this is fine, it’s also OK to introduce a semi “breaking change” once in a while, at the worst case someone won’t have their custom background color and will have to fix the code so not a big deal. Perhaps it’s even better for the sake of code clarity for the future.